index.js

```diff
@@ -690,13 +690,23 @@ Lexer.prototype = {
   code: function() {
     var captures;
     if (captures = /^(!?=|-)[ \t]*([^\n]+)/.exec(this.input)) {
-      this.consume(captures[0].length);
       var flags = captures[1];
-      captures[1] = captures[2];
-      var tok = this.tok('code', captures[1]);
+      var code = captures[2];
+      var parsed, shortened = 0;
+      if (this.interpolated) {
+        try {
+          parsed = characterParser.parseUntil(code, ']');
```

Comment:
I did not want to use `parseMax`, since consider this case:

```jade
p #[strong= ['a', 'b', 'c'][index]};
```

We could throw an error about "bracket mismatch,"  but that doesn't feel intuitive. One would expect the lexer to throw "no ending bracket found" since that is the most direct error.

And for this reason, I wrote ForbesLindesay/character-parser#12, which will allow `['a', 'b', 'c'][index]};` to be the source as long as there is a trailing `]`, and throw the correct error if not.